fastlane/lanes/release.rb

```diff
@@ -218,6 +219,49 @@ def create_release_backmerge_pr(version_to_merge:, next_version:)
   UI.user_error!(error_message)
 end
 
+def freeze_milestone_and_move_assigned_prs_to_next_milestone(
+  milestone_to_freeze:,
+  next_milestone:,
+  github_repository: GITHUB_REPO
+)
+  begin
+    # Move PRs to next milestone
+    moved_prs = update_assigned_milestone(
+      repository: github_repository,
+      from_milestone: milestone_to_freeze,
+      to_milestone: next_milestone,
+      comment: "Version `#{milestone_to_freeze}` has entered code-freeze. The milestone of this PR has been updated to `#{next_milestone}`."
+    )
+
+    # Add ❄️ marker to milestone title to indicate we entered code-freeze
+    set_milestone_frozen_marker(
+      repository: github_repository,
+      milestone: milestone_to_freeze
+    )
```

Comment:
I considered providing two methods with dedicated `begin rescue end` but decided to ask your thoughts on this first.

It might be overkill given the two steps always run together. 

One alternative might be to wrap each call in its own `begin rescue end` block, so that we can give a clear error message when one or the other fails.

Reply:
Actually one more strong reason to keep those 2 together is that they have to be run in that specific order and not the other way around, otherwise that would silently fail.

The reason for that is that if you rename the milestone first (from "X.Y" to "X.Y ❄️"), the search API we use for `update_assigned_milestone to search for opened PRs targeting said milestone will return an empty list and fail to find any.
From my prior testing when I noticed this back in the Tumblr repo, I think that might be because GitHub will take a second to update its internal database and search index to account for the new name, as if you retry the same search query a couple seconds later it will find the PRs again (which is what makes me think about a database index update taking some time). I didn't dig into this too much back then so maybe the real reason is something else, but in any case I know that doing the search query to `update_assigned_milestone` first and only then rename the milestone via `set_milestone_frozen_marker` avoids the problem altogether.

(All that will definitively be worth a code comment to be left in the implementation of the action once we move this all to `release-toolkit` at some point…)

Reply:
👍 added a note here in the meantime 9705d7ac